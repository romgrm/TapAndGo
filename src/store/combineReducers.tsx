import { combineReducers } from 'redux';
import { counterReducer } from './reducer';
import stationReducer from './stations/reducerStation'


export const rootReducer = combineReducers({
  counter: counterReducer,
  stations: stationReducer
});

export type RootState = ReturnType<typeof rootReducer>;