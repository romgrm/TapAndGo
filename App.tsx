import { StatusBar } from "expo-status-bar";
import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { MapScreen } from "./src/screens/mapScreen";
import { InformationsScreen } from "./src/screens/informationsScreen";
import stationsContainer from "./src/components/stationsContainer";
import { Provider } from "react-redux";
import store from "./src/store/store";

const Stack = createNativeStackNavigator();

export default function App() {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="Stations" component={stationsContainer}/>
          {/* <Stack.Screen name="Home" component={MapScreen} /> */}
          <Stack.Screen name="Infos" component={InformationsScreen} />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
