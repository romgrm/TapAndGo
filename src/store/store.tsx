import { createStore, applyMiddleware, combineReducers, Store } from "redux";
import { counterReducer } from "./reducer";
import stationReducer from "./stations/reducerStation";
import thunk from "redux-thunk";
import { rootReducer } from "./combineReducers";



const middleware = applyMiddleware(thunk);

const store = createStore(rootReducer, middleware);

export default store;
