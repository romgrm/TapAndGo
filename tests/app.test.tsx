import React from "react";
import renderer from "react-test-renderer";
import store from "../src/store/store";
import { Provider, useDispatch } from "react-redux";
import configureStore from "redux-mock-store";
import { render, cleanup, fireEvent } from "@testing-library/react-native";
import App from "../App";
import RootNavigation from "../src/navigation/rootNavigation";
import "react-native-gesture-handler/jestSetup";
import { NavigationContainer } from "@react-navigation/native";

afterEach(cleanup);

jest.mock("react-native-reanimated", () => {
  const Reanimated = require("react-native-reanimated/mock");

  // The mock for `call` immediately calls the callback which is incorrect
  // So we override it with a no-op
  Reanimated.default.call = () => {};

  return Reanimated;
});

// Silence the warning: Animated: `useNativeDriver` is not supported because the native animated module is missing
jest.mock("react-native/Libraries/Animated/src/NativeAnimatedHelper");

// jest.mock('react-navigation', () => ({
//     withNavigation: (Component: any) => (props:any) => (
//       <Component navigation={{ navigate: jest.fn() }} {...props} />
//     ),
//     SafeAreaView: ({ children }:any) => <>{children}</>,
//   }));
describe("<App />", () => {
  // configureStore expects a list of middlewares
  const initialState = {
    isLoading: false,
    station: [
      {
        station: "nantes",
      },
      {
        station: "rennes",
      },
    ],
    error: "error",
  };
  const mockStore = configureStore([]);
  it("should render", async () => {
    const storeMock = mockStore({ station: ["fake", "fake2"] });
    // const apppTest = render(
    //   <Provider store={storeMock}>
    //     <NavigationContainer>
    //       <RootNavigation />
    //     </NavigationContainer>
    //   </Provider>
    // );
    // const tree = getByTestId('test')
    await expect(
      renderer.create(
        <Provider store={storeMock}>
          <App />
        </Provider>
      )
    ).toMatchSnapshot();
  });

  // it('should dispatch increment action', () => {
  //   const store = mockStore({count: 5});
  //   const rendered = render(
  //     <Provider store={store}><Counter /></Provider>
  //   );
  //   const buttonComponent = rendered.getByTestId('button');

  //   fireEvent(buttonComponent, 'press');

  //   // This will return all actions dispatched on this store
  //   const actions = store.getActions();
  //   expect(actions.length).toBe(1);
  //   expect(actions[0].type).toEqual('INCREMENT');
  // });
});
// describe("<App />", () => {
//   const dispatch = jest.fn();
//   store.dispatch = dispatch;
//   const fetchData = jest.fn();

//   beforeEach(() => {
//     render(
//       <Provider store={store}>
//         <App />
//       </Provider>
//     );
//   });
//   it("has 1 child", () => {
//     const tree = renderer.create(<App />).toJSON();
//     expect(tree).toBe(1);
//   });
// });
