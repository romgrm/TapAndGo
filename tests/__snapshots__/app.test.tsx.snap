// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render 1`] = `null`;

exports[`<MapScreen /> renders correctly 1`] = `null`;
